test_util/aws.py

```diff
@@ -0,0 +1,113 @@
+        self.resource('cloudformation').create_stack(
+            StackName=name,
+            TemplateURL=template_url,
+            DisableRollback=True,
+            TimeoutInMinutes=deploy_timeout,
+            Capabilities=['CAPABILITY_IAM'],
+            Parameters=cf_parameters)
+        return CFStack(name, self)
+
+
+class CFStack():
+    def __init__(self, stack_name, boto_wrapper):
+        self.boto_wrapper = boto_wrapper
+        self.stack = self.boto_wrapper.resource('cloudformation').Stack(stack_name)
+
+    def wait_for_stack_creation(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_create_complete').\
+            wait(StackName=self.stack.stack_name)
+
+    def wait_for_stack_deletion(self):
+        self.boto_wrapper.client('cloudformation').get_waiter('stack_delete_complete').\
+            wait(StackName=self.stack.stack_name)
+
+
+class DcosCfSimple(CFStack):
+    @classmethod
+    def create(cls, stack_name, template_url, public_agents, private_agents,
+               admin_location, key_pair_name, boto_wrapper):
+        parameters = {
+            'KeyName': key_pair_name,
+            'AdminLocation': admin_location,
+            'PublicSlaveInstanceCount': str(public_agents),
+            'SlaveInstanceCount': str(private_agents)}
+        boto_wrapper.create_stack(stack_name, template_url, parameters)
+        return cls(stack_name, boto_wrapper)
+
+    def delete(self):
+        logger.info('Starting deletion of CF stack')
+        # boto stacks become unusable after deletion (e.g. status/info checks) if name-based
+        self.stack = self.boto_wrapper.resource('cloudformation').Stack(self.stack.stack_id)
+        self.stack.delete()
+        bucket_id = self.stack.Resource('ExhibitorS3Bucket').physical_resource_id
+        s3 = self.boto_wrapper.resource('s3')
+        bucket = s3.Bucket(bucket_id)
+        logger.info('Starting bucket {} deletion'.format(bucket))
+        all_objects = bucket.objects.all()
+        obj_count = len(list(all_objects))
+        if obj_count > 0:
+            assert obj_count == 1, 'Expected one object in Exhibitor S3 bucket but found: ' + str(obj_count)
+            logger.info('Trying to delete object from bucket: {}'.format(repr(all_objects[0])))
+            all_objects[0].delete()
+        logger.info('Trying deleting bucket {} itself'.format(bucket))
+        bucket.delete()
+        logger.info('Delete successfully triggered for {}'.format(self.stack.stack_name))
+
+    def get_tag_instances(self, tag):
+        return self.boto_wrapper.client('ec2').describe_instances(
+            Filters=[
+                {'Name': 'tag-value', 'Values': [self.stack.stack_name]},
+                {'Name': 'tag-value', 'Values': [tag]}])['Reservations'][0]['Instances']
+
+    def get_master_ips(self):
+        instances = self.get_tag_instances('MasterServerGroup')
+        return [Host(i['PrivateIpAddress'], i['PublicIpAddress']) for i in instances]
```

Comment:
Sure, but it will be less-than-elegant function because it needs to know when there isnt a PublicIpAddress key. E.G.
```
hosts = []
for i in instances:
    host_i = Host(None, None)
    if 'PrivateIpAddress' in i:
         host.private_ip = i[..]
    if 'PublicIpAddress' in i:
        host.public_ip = i[...]
    hosts.append(hosts_i)
return hosts
```

Seemed to me that the one-liner comprehension was better than getting if's involved